lib/components/global/tooltip/Tooltip.js

```diff
@@ -23,15 +23,56 @@ var Tooltip = React.createClass({
 		orientation : React.PropTypes.string,
 		color       : React.PropTypes.string,
 		background  : React.PropTypes.string,
-		width       : React.PropTypes.oneOfType([
-			React.PropTypes.string,
-			React.PropTypes.number
-		]),
-		height: React.PropTypes.oneOfType([
-			React.PropTypes.string,
-			React.PropTypes.number
-		]),
-		keepOpen: React.PropTypes.bool
+		width       : React.PropTypes.string,
+		height      : React.PropTypes.string,
+		keepOpen    : React.PropTypes.bool
+	},
+
+	unitRegExp  : new RegExp(/^\d+(px|em|\%|cm|mm|in|pt|pc|ch|vh|vw|vmin|vmax)$/),
+	digitRegExp : new RegExp(/\d+/),
+
+	setWidth: function(width, orientation, styles) {
+		try {
+
+			if (! this.unitRegExp.test(width)) {
+				let newWidth = width.match(this.digitRegExp)[0] + 'px';
```

Comment:
An explicit exception can be thrown here to clearly show where the exception is being thrown.